Devops 1440 update ruff 0.4.4

bio-boris opened the pull request:
* Pin ruff 0.4.4
* Force precommit to format your code
* Fix issues with pyproject.toml using old version of configs

Commit:
Refactor ruff

Commit:
Refactor ruff

ialarmedalien commented on pyproject.toml:
```diff
@@ -65,14 +48,22 @@ exclude = [
+target-version = "py312"
 
-# Same as Black.
+[tool.black]
+# Don't format the installed clients or the SDK-generated files.
+extend-exclude = '''
+import_specifications/clients/authclient.py
+'''
```
this isn't needed with the ruff formatter

bio-boris committed:
Update pyproject.toml

bio-boris commented:
```
there’s an extra segment in the pyproject.toml file that can be removed, otherwise looks fine.
```
Removed that section, squashed it, and merging it in!
